### ISSUE TITLE
Doesn't recognize source generated code

### ISSUE THREAD

**Issue by trampster** [dotnet/vscode-csharp#6182]
Type: <b>Bug</b>

## Issue Description ##
Generated code isn't recognized by the c# vscode extension. Generated types/methods are displayed with red underlines and hovering tells you it can't find the method

## Steps to Reproduce ##
* Checkout https://github.com/trampster/SqlSrcGen
* open with vscode with the c# extension installed
* navigate to IntegrationTests
* Build with dotnet build (it builds fine)
* open NullableTests.cs (notice methods calls to generated _database methods are underlined in read.)

## Expected Behavior ##
* the c# vscode extension recongises generated types/methods
* they are not underlined in red
* autocomplete/intelisense works on them.
* I can navigate to generated methods.

## Actual Behavior ##
* the generated Database type and all it's methods are not recognized
* they are underlined in red
* hovering shows errors as follows:

> The type or namespace name 'Database' could not be found (are you missing a using directive or an assembly reference?)RoslynCS0246

### C# log ###
<details>Failed to find dotnet info from path, falling back to acquire runtime via ms-dotnettools.vscode-dotnet-runtime
Found dotnet version 7.0.10. Minimum required version is 7.0.100.
Dotnet path: /home/daniel/.config/Code/User/globalStorage/ms-dotnettools.vscode-dotnet-runtime/.dotnet/7.0.10/dotnet
Activating C# standalone...
[Info  - 21:28:56] [Program]Language server initialized
[Info  - 21:28:57] [LanguageServerProjectSystem]Loaded MSBuild at /usr/share/dotnet/sdk/7.0.400
[Info  - 21:28:57] [LanguageServerProjectSystem]Loading /home/daniel/Work/SqlSrcGen/SqlSrcGen.sln...
[Info  - 21:28:59] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/IntegrationTests/IntegrationTests.csproj
[Info  - 21:28:59] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/SqlSrcGen/SqlSrcGen.csproj
[Info  - 21:28:59] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/SqlSrcGen.Runtime/SqlSrcGen.Runtime.csproj
[Info  - 21:28:59] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Tests/Tests.csproj
[Info  - 21:28:59] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Playground/Playground.csproj
[Info  - 21:28:59] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Benchmarks/Benchmarks.csproj
[Info  - 21:28:59] [LanguageServerProjectSystem]Completed (re)load of all projects in 00:00:02.1758626
[Info  - 21:30:08] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/IntegrationTests/IntegrationTests.csproj
[Info  - 21:30:08] [LanguageServerProjectSystem]Completed (re)load of all projects in 00:00:00.3324871
[Info  - 21:30:42] Shutting down
Failed to find dotnet info from path, falling back to acquire runtime via ms-dotnettools.vscode-dotnet-runtime
Found dotnet version 7.0.10. Minimum required version is 7.0.100.
Dotnet path: /home/daniel/.config/Code/User/globalStorage/ms-dotnettools.vscode-dotnet-runtime/.dotnet/7.0.10/dotnet
Activating C# standalone...
[Info  - 21:30:43] [Program]Language server initialized
[Info  - 21:30:44] [LanguageServerProjectSystem]Loaded MSBuild at /usr/share/dotnet/sdk/7.0.400
[Info  - 21:30:44] [LanguageServerProjectSystem]Loading /home/daniel/Work/SqlSrcGen/SqlSrcGen.sln...
[Info  - 21:30:46] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Benchmarks/Benchmarks.csproj
[Info  - 21:30:46] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Tests/Tests.csproj
[Info  - 21:30:46] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/IntegrationTests/IntegrationTests.csproj
[Info  - 21:30:46] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/SqlSrcGen.Runtime/SqlSrcGen.Runtime.csproj
[Info  - 21:30:46] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Playground/Playground.csproj
[Info  - 21:30:46] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/SqlSrcGen/SqlSrcGen.csproj
[Info  - 21:30:46] [LanguageServerProjectSystem]Completed (re)load of all projects in 00:00:01.9939267
[Info  - 21:32:02] Shutting down
Failed to find dotnet info from path, falling back to acquire runtime via ms-dotnettools.vscode-dotnet-runtime
Found dotnet version 7.0.10. Minimum required version is 7.0.100.
Dotnet path: /home/daniel/.config/Code/User/globalStorage/ms-dotnettools.vscode-dotnet-runtime/.dotnet/7.0.10/dotnet
Activating C# standalone...
[Info  - 21:32:03] [Program]Language server initialized
[Info  - 21:32:04] [LanguageServerProjectSystem]Loaded MSBuild at /usr/share/dotnet/sdk/7.0.400
[Info  - 21:32:04] [LanguageServerProjectSystem]Loading /home/daniel/Work/SqlSrcGen/SqlSrcGen.sln...
[Info  - 21:32:05] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/SqlSrcGen.Runtime/SqlSrcGen.Runtime.csproj
[Info  - 21:32:05] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/IntegrationTests/IntegrationTests.csproj
[Info  - 21:32:05] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/SqlSrcGen/SqlSrcGen.csproj
[Info  - 21:32:06] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Playground/Playground.csproj
[Info  - 21:32:06] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Tests/Tests.csproj
[Info  - 21:32:06] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/Benchmarks/Benchmarks.csproj
[Info  - 21:32:06] [LanguageServerProjectSystem]Completed (re)load of all projects in 00:00:01.9900941
[Info  - 21:38:29] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/IntegrationTests/IntegrationTests.csproj
[Info  - 21:38:29] [LanguageServerProjectSystem]Completed (re)load of all projects in 00:00:00.3462084
[Info  - 21:39:16] [LanguageServerProjectSystem]Successfully completed load of /home/daniel/Work/SqlSrcGen/IntegrationTests/IntegrationTests.csproj
[Info  - 21:39:16] [LanguageServerProjectSystem]Completed (re)load of all projects in 00:00:00.1742078
Failed to find dotnet info from path, falling back to acquire runtime via ms-dotnettools.vscode-dotnet-runtime
Found dotnet version 7.0.10. Minimum required version is 7.0.100.
e</details>

### C# LSP Trace Logs ###
<details>Post the output from Output-->C# LSP Trace Logs here.  Requires `dotnet.server.trace` to be set to `Trace`</details>

## Environment information ##

**VSCode version**: 1.81.1
**C# Extension**: 2.0.376
**Using OmniSharp**: false

<details><summary>Dotnet Information</summary>
.NET SDK:
 Version:   7.0.400
 Commit:    73bf45718d

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.400/

Host:
  Version:      7.0.10
  Architecture: x64
  Commit:       a6dbb800a4

.NET SDKs installed:
  5.0.408 [/usr/share/dotnet/sdk]
  6.0.413 [/usr/share/dotnet/sdk]
  7.0.400 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.17 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.21 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.17 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.21 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|csharp|ms-dotnettools|2.0.376|ms-dotnettools.csharp-2.0.376-linux-x64|
|docomment|k--kato|1.0.0|k--kato.docomment-1.0.0|
|dotnet-test-explorer|formulahendry|0.7.8|formulahendry.dotnet-test-explorer-0.7.8|
|EditorConfig|EditorConfig|0.16.4|editorconfig.editorconfig-0.16.4|
|hexeditor|ms-vscode|1.9.12|ms-vscode.hexeditor-1.9.12|
|markdown-preview-enhanced|shd101wyy|0.6.8|shd101wyy.markdown-preview-enhanced-0.6.8|
|pimp-my-avalonia|PrzemyslawOnak|0.0.1|przemyslawonak.pimp-my-avalonia-0.0.1|
|plantuml|jebbs|2.17.5|jebbs.plantuml-2.17.5|
|remote-explorer|ms-vscode|0.4.1|ms-vscode.remote-explorer-0.4.1|
|remote-ssh|ms-vscode-remote|0.102.0|ms-vscode-remote.remote-ssh-0.102.0|
|remote-ssh-edit|ms-vscode-remote|0.86.0|ms-vscode-remote.remote-ssh-edit-0.86.0|
|ShowTestResults|EffectiveLabs|0.4.0|EffectiveLabs.ShowTestResults-0.4.0|
|vscode-dotnet-runtime|ms-dotnettools|1.7.0|ms-dotnettools.vscode-dotnet-runtime-1.7.0|
|vsliveshare|ms-vsliveshare|1.0.5877|ms-vsliveshare.vsliveshare-1.0.5877|
|vsliveshare-audio|ms-vsliveshare|0.1.93|ms-vsliveshare.vsliveshare-audio-0.1.93|
|vsliveshare-pack|ms-vsliveshare|0.4.0|ms-vsliveshare.vsliveshare-pack-0.4.0|;

</details>

Extension version: 2.0.376
VS Code version: Code 1.81.1 (6c3e3dba23e8fadc360aed75ce363ba185c49794, 2023-08-09T22:18:39.991Z)
OS version: Linux x64 4.15.0-213-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz (8 x 2233)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|0, 0, 0|
|Memory (System)|31.33GB (26.27GB free)|
|Process Argv|--unity-launch --crash-reporter-id eba97097-b99b-4a82-95ac-9c25288c319e|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vstes627:30244334
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vscccc:30803845
282f8724:30602487
f6dab269:30613381
2i9eh265:30646982
showlangstatbar:30737416
03d35959:30757346
ecj1e332:30736112
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
gsofa:30804715
pythonnosmt12:30797651
pythonidxptcf:30805731
pythonnoceb:30805159
dsvsc013:30795093
dsvsc014:30804076
diffeditorv2:30812749

```

</details>

**Comment by trampster** [comment]
[extensionsLogs.zip](https://github.com/dotnet/vscode-csharp/files/12394884/extensionsLogs.zip)

**Comment by glen-84** [comment]
Could be a duplicate of #6053?

**Comment by trampster** [comment]
Could be, mine isn't a partial class, but root cause could be the same.

![SourceGeneratorVscodecSharpExtIssue](https://github.com/dotnet/vscode-csharp/assets/2063633/8e77050e-96d5-4737-a166-397baa452cb1)

**Comment by arkalyanms** [comment]
I believe this and https://github.com/dotnet/vscode-csharp/issues/6053 should be covered by https://github.com/dotnet/vscode-csharp/pull/5858.

**Comment by trampster** [comment]
I notice this is in the September milestone but it's October. Looks like the linked fixed #5858 isn't merged yet (has merge conflicts with no changes since Aug 22).

Does the milestone need to be changed?

**Comment by jasonmalinowski** [comment]
@trampster #5858 wouldn't actually help in this case; I do plan on digging into this and other reports later this week. (Which is September...ish. :smile:)
